core/capabilities/ccip/ccipevm/extraargscodec.go

```diff
@@ -1,16 +1,60 @@
 package ccipevm
 
 import (
+	"fmt"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
 	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
 )
 
+const (
+	EVMExtraArgsKey     = "gasLimit"
+	SVMComputeUnitKey   = "ComputeUnits"
+	SVMAccountBitmapKey = "AccountIsWritableBitmap"
+	SVMAllowOOEKey      = "AllowOutOfOrderExecution"
+	SVMTokenReceiverKey = "TokenReceiver"
+	SVMAccountListKey   = "Accounts"
+)
+
 type ExtraArgsCodec struct{}
 
 func NewExtraArgsCodec() ExtraArgsCodec {
 	return ExtraArgsCodec{}
 }
 
 func (ExtraArgsCodec) DecodeExtraData(extraArgs cciptypes.Bytes, sourceChainSelector cciptypes.ChainSelector) (map[string]any, error) {
-	// Not implemented and not return error
-	return nil, nil
+	family, err := chainsel.GetSelectorFamily(uint64(sourceChainSelector))
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode extra data, %w", err)
+	}
+
+	switch family {
+	case chainsel.FamilyEVM:
+		gas, err1 := decodeExtraArgsV1V2(extraArgs)
+		if err1 != nil {
+			return nil, fmt.Errorf("failed to decode EVM extra data, %w", err)
+		}
+
+		return map[string]any{
+			EVMExtraArgsKey: gas,
+		}, nil
+
+	case chainsel.FamilySolana:
```

Comment:
The switch on family here is on the source chain selector, meaning this case applies when source was Solana chain.
Thus decoding this requires Solana decoder.
But the implementation is using EVM decoder.